controller/src/beerocks/master/son_master_thread.cpp

```diff
@@ -373,6 +373,97 @@ bool master_thread::handle_cmdu_1905_autoconfiguration_search(Socket *sd,
     return son_actions::send_cmdu_to_agent(sd, cmdu_tx);
 }
 
+/**
+ * @brief Encrypt the config data using AES and add to the WSC M2 TLV
+ *        The encrypted data length is the config data length padded to 16 bytes boundary.
+ *
+ * @param[in] m2 WSC M2 TLV
+ * @param[in] config_data config data in network byte order (swapped)
+ * @param[in] authkey 32 bytes calculated authentication key
+ * @param[in] keywrapkey 16 bytes calculated key wrap key
+ * @return true on success
+ * @return false on failure
+ */
+bool master_thread::autoconfig_wsc_add_m2_encrypted_settings(
+    std::shared_ptr<ieee1905_1::tlvWscM2> m2, WSC::cConfigData &config_data, uint8_t authkey[32],
+    uint8_t keywrapkey[16])
+{
+    size_t len             = (config_data.getLen() + 16) & ~0xFU; // Align to 16 bytes boundary
```

Comment:
I may have a problem with my math-fu, but align to 16 bytes should add just 15 to it, no? Otherwise, if it already is a multiple of 16, you add redundant padding.